quickwit-search/src/root.rs

```diff
@@ -232,16 +239,21 @@ fn jobs_to_fetch_docs_request(
     partial_hits_map: &mut HashMap<String, Vec<PartialHit>>,
     jobs: &[Job],
 ) -> FetchDocsRequest {
+    let partial_hits_map_size = partial_hits_map.len();
     let partial_hits = jobs
         .iter()
-        .map(|job| partial_hits_map.remove(&job.split_id).unwrap())
+        .filter_map(|job| partial_hits_map.remove(&job.split_id))
         .flatten()
         .collect_vec();
+    debug_assert_eq!(partial_hits_map_size - partial_hits_map.len(), jobs.len());
```

Comment:
Why not :
```suggestion
    assert_eq!(partial_hits.len(), jobs.len());
```

Reply:
Because `partial_hits_map` value is not flatten. Partial hits might have different size from jobs. The correct one is checking number of removed key.